adapters/repos/db/shard_write_batch_objects.go

```diff
@@ -151,11 +151,24 @@ func (ob *objectsBatcher) storeSingleBatchInLSM(ctx context.Context,
 	}
 
 	wg := &sync.WaitGroup{}
+	concurrencyLimit := make(chan struct{}, _NUMCPU)
```

Comment:
Interesting concept 👍

Would `errgroup` with `SetLimit` or fixed number of go routines with objects read from channel work here as well?

Reply:
It's actually ChatGPT that came up with this. 🤣  But I like it. It's idiomatic Go, quite concise, and works great.

I think an `errgroup` would work just as well, yes. The reason I didn't go for one is because we're currently handling each error individually whereas with an `errgroup` the assumption is that you want to handle all errors together.